pkg/process/build/build.go

```diff
@@ -20,158 +20,155 @@ type Builder interface {
-		if err != nil {
-			return errors.Wrapf(err, "Image tag failed")
-		}
-		metatags := make(map[string]string)
-		for i, tag := range tags {
-			metatags[t[i]] = tag
-		}
-		if !cfg.NoPush {
-			err = builder.Push(ctx, buildResult, tags)
-			manifest, err := pushRegistry.GetImageInfo(ctx, buildConfig.DockerRepository, t[0])
-			if err == nil {
-				imageConfig, err := pushRegistry.GetImageConfig(ctx, buildConfig.DockerRepository, manifest.Digest)
-				if err == nil {
-					tracer.AddImageMetadata(trace.DeployableImage{
-						Type:         "deployableImage",
-						Digest:       manifest.Digest,
-						Name:         buildConfig.DockerRepository,
-						Tags:         metatags,
-						ImageConfig:  imageConfig,
-						NexusSHA1:    deliverable.SHA1,
-						Dependencies: dependencyMetadata,
-					})
-				} else {
-					logrus.Warnf("Unable to find information about %s:%s. Got error: %s",
-						buildConfig.DockerRepository, imageInfo.Digest, err)
-				}
-			} else {
-				logrus.Warnf("Unable to find information about %s:%s. Got error: %s",
-					buildConfig.DockerRepository, t[0], err)
-			}
-		}
-		return err
+	baseImage := runtime.BaseImage{
+		DockerImage: runtime.DockerImage{
+			Tag:        completeBaseImageVersion,
+			Repository: baseImageSpec.BaseImage,
+			Registry:   cfg.DockerSpec.GetInternalPullRegistryWithoutProtocol(),
+		},
+		ImageInfo: imageInfo,
 	}
+	return baseImage, nil
+}
+
+func buildDockerImage(buildConfig docker.BuildConfig, ctx context.Context, cfg *config.Config, layerBuilder Builder) (*LayerProvider, error) {
+
+	if cfg.NexusIQReportUrl != "" {
+		buildConfig.Labels["no.skatteetaten.aurora.nexus-iq-report-url"] = cfg.NexusIQReportUrl
+	}
+
+	baseImageLayers, err := layerBuilder.Pull(ctx, buildConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "There was an error with the pull operation.")
+	}
+
+	logrus.Info("Docker context ", buildConfig.BuildFolder)
+	return layerBuilder.Build(buildConfig, baseImageLayers)
+}
+func pushImage(ctx context.Context, cfg *config.Config, buildResult *LayerProvider, layerBuilder Builder, tags []string) error {
+	if cfg.NoPush {
+		return nil
+	}
+
+	err := layerBuilder.Push(ctx, buildResult, tags)
+	if err != nil {
+		return errors.Wrapf(err, "Image push failed")
+	}
+
 	return nil
 }
 
-func checkAllTagsForOverwrite(ctx context.Context, dockerBuildConfigArray []docker.BuildConfig, pushRegistry docker.Registry, cfg *config.Config) error {
+func sendImageInfoToSporingsLogger(ctx context.Context, cfg *config.Config, serviceName string,
+	dockerRegistry docker.Registry, sporingsLoggerClient trace.Trace, auroraVersion *runtime.AuroraVersion, shortTags []string) error {
+	if cfg.NoPush {
+		logrus.Info("sendImageInfoToSporingsLogger NoPush")
+		return nil
+	}
+
+	imageInfo, err := dockerRegistry.GetImageInfo(ctx, serviceName, shortTags[0])
+	if err != nil {
+		return errors.Wrapf(err, "Unable to GetImageInfo ")
+	}
+	logrus.Info("sendImageInfoToSporingsLogge")
```

Comment:
Hva med

    Sending image info to sporingslogger

Men kanskje legge på noe mer som gjør at vi kan hente det ut også? Hva med digest?